test/integration/DelayedRecovery.js

```diff
@@ -0,0 +1,404 @@
+
+  const recoveryModule = new auxiliaryWeb3.eth.Contract(recoveryModuleAbi, recoveryModuleAddressProxy);
+
+  return recoveryModule;
+}
+
+function hashRecoveryModuleDomainSeparator(recoveryModuleAddress) {
+  return auxiliaryWeb3.utils.keccak256(
+    auxiliaryWeb3.eth.abi.encodeParameters(
+      ['bytes32', 'address'],
+      [RECOERY_MODULE_DOMAIN_SEPARATOR_TYPEHASH, recoveryModuleAddress]
+    )
+  );
+}
+
+function hashRecoveryModuleRecoveryStruct(structTypeHash, prevOwner, oldOwner, newOwner) {
+  return auxiliaryWeb3.utils.keccak256(
+    auxiliaryWeb3.eth.abi.encodeParameters(
+      ['bytes32', 'address', 'address', 'address'],
+      [structTypeHash, prevOwner, oldOwner, newOwner]
+    )
+  );
+}
+
+function hashRecoveryModuleRecovery(recoveryModuleAddress, structTypeHash, prevOwner, oldOwner, newOwner) {
+  const recoveryStructHash = hashRecoveryModuleRecoveryStruct(structTypeHash, prevOwner, oldOwner, newOwner);
+
+  const domainSeparatorHash = hashRecoveryModuleDomainSeparator(recoveryModuleAddress);
+
+  return auxiliaryWeb3.utils.soliditySha3(
+    { t: 'bytes1', v: '0x19' },
+    { t: 'bytes1', v: '0x01' },
+    { t: 'bytes32', v: domainSeparatorHash },
+    { t: 'bytes32', v: recoveryStructHash }
+  );
+}
+
+function signRecovery(recoveryModuleAddress, structTypeHash, prevOwner, oldOwner, newOwner, recoveryOwnerPrivateKey) {
+  const recoveryHash = hashRecoveryModuleRecovery(recoveryModuleAddress, structTypeHash, prevOwner, oldOwner, newOwner);
+
+  const signature = EthUtils.ecsign(EthUtils.toBuffer(recoveryHash), EthUtils.toBuffer(recoveryOwnerPrivateKey));
+
+  return {
+    recoveryHash,
+    signature
+  };
+}
+
+function signInitiateRecovery(recoveryModuleAddress, prevOwner, oldOwner, newOwner, recoveryOwnerPrivateKey) {
+  return signRecovery(
+    recoveryModuleAddress,
+    INITIATE_RECOVERY_STRUCT_TYPEHASH,
+    prevOwner,
+    oldOwner,
+    newOwner,
+    recoveryOwnerPrivateKey
+  );
+}
+
+function signExecuteRecovery(recoveryModuleAddress, prevOwner, oldOwner, newOwner, recoveryOwnerPrivateKey) {
+  return signRecovery(
+    recoveryModuleAddress,
+    EXECUTE_RECOVERY_STRUCT_TYPEHASH,
+    prevOwner,
+    oldOwner,
+    newOwner,
+    recoveryOwnerPrivateKey
+  );
+}
+
+function advanceBlock() {
+  return new Promise((resolve, reject) => {
+    auxiliaryWeb3.currentProvider.send(
+      {
+        jsonrpc: '2.0',
+        method: 'evm_mine',
+        id: new Date().getTime()
+      },
+      (err) => {
+        if (err) {
+          return reject(err);
+        }
+
+        const newBlockHash = auxiliaryWeb3.eth.getBlock('latest').hash;
+
+        return resolve(newBlockHash);
+      }
+    );
+  });
+}
+
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+async function waitBlockNumber(blockNumber) {
+  let currentBlockNumber = await auxiliaryWeb3.eth.getBlockNumber();
+
+  while (currentBlockNumber < blockNumber) {
+    await sleep(2000);
```

Comment:
this probably needs to be fixed later... ok for now to avoid creating a blocker.

Reply:
Could you please share your proposal?